Source/GSOrderedSet.m

```diff
@@ -0,0 +1,399 @@
+  for (i = 0; i < c; i++)
+    {
+      id obj = objs[i];
+      GSIArrayItem item;
+      
+      if (objs[i] == nil)
+	{
+	  DESTROY(self);
+	  [NSException raise: NSInvalidArgumentException
+		      format: @"Tried to init set with nil value"];
+	}
+
+      item.obj = obj;
+      if(![self containsObject: obj])
+	{
+	  GSIArrayAddItem(&array, item);
+	  RETAIN(obj);
+	}
+    }
+  return self;
+}
+
+@end
+
+@implementation GSMutableOrderedSet
+
++ (void) initialize
+{
+  if (self == [GSMutableOrderedSet class])
+    {
+      GSObjCAddClassBehavior(self, [GSOrderedSet class]);
+    }
+}
+
+- (void) addObject: (id)anObject
+{
+  GSIArrayItem item;
+
+  if (anObject == nil)
+    {
+      [NSException raise: NSInvalidArgumentException
+		  format: @"Tried to add nil to set"];
+    }
+
+  if([self containsObject: anObject] == NO)
+    {
+      item.obj = anObject;
+      GSIArrayAddItem(&array, item);
+      RETAIN(anObject);
+      _version++;
+    }
+}
+
+- (void) insertObject: (id)object atIndex: (NSUInteger)index
+{
+  GSIArrayItem item;
+  if([self containsObject: object] == NO)
+    {
+      item.obj = object;
+      GSIArrayInsertItem(&array, item, index);
+      RETAIN(object);
+      _version++;
+    }
+}
+
+- (void) removeObjectAtIndex: (NSUInteger)index
+{
+  _version++;
+  GSIArrayRemoveItemAtIndex(&array, index);
+}
+
+- (void) replaceObjectAtIndex: (NSUInteger)index
+		   withObject: (id)obj
+{
+  [self removeObjectAtIndex: index];
+  [self insertObject: obj atIndex: index];
+}
+
+- (id) init
+{
+  return [self initWithCapacity: 0];
+}
+
+/* Designated initialiser */
+- (id) initWithCapacity: (NSUInteger)cap
+{
+  GSIArrayInitWithZoneAndCapacity(&array, [self zone], cap);
+  return self;
+}
+
+- (id) initWithObjects: (const id*)objects
+		 count: (NSUInteger)count
+{
+  NSUInteger i = 0;
+
+  // Init and fill set
+  self = [self initWithCapacity: count];
+  if(self != nil)
+    {
+      for(i = 0; i < count; i++)
+	{
+	  id	anObject = objects[i];	  
+	  if (anObject == nil)
+	    {
+	      NSLog(@"Tried to init an orderedset with a nil object");
+	      continue;
+	    }
+	  else
+	    {
+	      GSIArrayItem item;
+	      if(![self containsObject: anObject])
+		{
+		  item.obj = anObject;
+		  GSIArrayAddItem(&array, item);
+		  RETAIN(anObject);
+		}
+	    }
+	}
+    }
+  return self;
+}
+
+- (BOOL) makeImmutable
+{
+  GSClassSwizzle(self, [GSOrderedSet class]);
+  return YES;
+}
+
+- (id) makeImmutableCopyOnFail: (BOOL)force
+{
+  GSClassSwizzle(self, [GSOrderedSet class]);
+  return self;
+}
+
+- (NSUInteger) countByEnumeratingWithState: (NSFastEnumerationState*)state
+                                   objects: (id*)stackbuf
+                                     count: (NSUInteger)len
+{
+  //state->mutationsPtr = (unsigned long *)&_version;
+  //return GSIMapCountByEnumeratingWithStateObjectsCount
+  //  (&map, state, stackbuf, len);
+  return [self count];
+}
+@end
+
+@interface	NSGOrderedSet : NSOrderedSet
```

Comment:
I googled for these two classes but could not find any reference to them. In NSArray things are different, there Richard needs to handle the old implementation of NSArray that made its way into a few NIB files that we still want to load.

Reply:
I had put them there because they were in NSArray... I wasn't sure what the use for them was.

Reply:
Removed